src/reason-parser/reason_pprint_ast.ml

```diff
@@ -6640,6 +6644,159 @@ class printer  ()= object(self:'self)
     match xs with
     | [x] -> x
     | xs -> makeBreakableList xs
+
+  method formatFunAppl ~jsxAttrs ~args ~funExpr () =
+    (* If there was a JSX attribute BUT JSX component wasn't detected,
+       that JSX attribute needs to be pretty printed so it doesn't get
+       lost *)
+    let maybeJSXAttr = List.map self#attribute jsxAttrs in
+
+    let categorizeFunApplArgs args =
```

Comment:
To address @chenglou's finding, I propose that we change this to detect when the final argument is a callback and no other argument is a callback.